.github/workflows/test_editable_pip_install.yml

```diff
@@ -0,0 +1,63 @@
+name: Test Editable Pip Install
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  editable_install:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04, macos-latest]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # This shouldn't be necessary, but for some reason it is on macos
+      - name: install dev_requirements.txt
+        if: ${{ matrix.os == 'macos-latest' }}
+        run: python3 -m pip install -r dev_requirements.txt
+
+      - name: Upgrade versions
+        run: |
+          python3 -m pip install --upgrade pip
+          python3 -m pip install setuptools==65.3.0
```

Comment:
I'm a little worried by this constant, and pip being unpinned - can we add a matrix entry for pip version and setuptools version, even if they only have one entry right now?

Reply:
That sounds reasonable.